src/opm/parser/eclipse/EclipseState/Schedule/Well/WellTestState.cpp

```diff
@@ -85,12 +100,25 @@ namespace Opm {
         return this->wells.size();
     }
 
-    std::vector<std::pair<std::string, WellTestConfig::Reason>> WellTestState::updateWell(const WellTestConfig& config, double sim_time) {
+    std::vector<std::pair<std::string, WellTestConfig::Reason>>
+    WellTestState::updateWell(const WellTestConfig& config,
+                              const std::vector<Well2>& wells_ecl,
+                              double sim_time) {
         std::vector<std::pair<std::string, WellTestConfig::Reason>> output;
 
         updateForNewWTEST(config);
 
         for (auto& well : this->wells) {
+            auto well_ecl = std::find_if(wells_ecl.begin(), wells_ecl.end(),
+                    [&well](const Well2& w)
+                    {
+                        return w.name() == well.name;
+                    });
+
+            // if the well is SHUT, we do not consider to test it
+            if (well_ecl->getStatus() != WellCommon::OPEN)
```

Comment:
What if `std::find_if()` returns `wells_ecl.end()`?

Reply:
It turned out that it is possible in the unit test, when we create `wells_ecl` manually.

Reply:
I guess @joakim-hove hinted that this should be improved to:
```
if (well_ecl == wells_ecl.end() || well_ecl->getStatus() != WellCommon::OPEN)
```

Reply:
It is a good point.  

I was thinking to update the code to be this way, then I think if it actually happens, either the way to use this function or the way the function itself needs to be re-designed. 

I will keep the throwing for now.